FileSpecs/Maps.md

```diff
@@ -0,0 +1,206 @@
+# Maps
+
+**Note:** This is **WIP**. Some data structures are incomplete, others may be incorrect.
+
+Maps are stored in `MAP_DATA.AMB`.  Each resource describes one
+in-game map, with 2D and 3D maps sharing much of the file structure.
+The files consist of the following sections:
+* Header
+* Event descriptions (e.g., popup messages, chests, locked doors, or transitions to other maps)
+* NPC references (not including the actual NPC descriptions, which are in `CHARDATA.AMB`)
+* A magic 9 byte sequence
+* **3D maps only**: LabInfo blocks (meaning currently unclear)
+* The actual map
+* **2D maps only**: An overlay tile map, drawn over the preceding map
+* A hotspot map, which maps map locations to event descriptions
+* NPC locations and/or movement routes
+
+
+## Map Header and Structure
+
+| Offset   | Format             | Name                 | Description                                                                      |
+|----------|--------------------|----------------------|----------------------------------------------------------------------------------|
+| `0`      | u8                 | magic number         | = `0xff`                                                                         |
+| `1`      | u8                 | magic number         | = `0x00`                                                                         |
+| `2`      | u8                 | magic number         | = `0x00`                                                                         |
+| `3`      | u8                 | tileset / background | 2D maps: tileset (`1` or `2`).  3D maps: background image (?) (`0x01`-`0x17`)    |
```

Comment:
In Ambermoon this would give the lab data subfile index for 3D maps.

Reply:
Good catch!  The numbers align perfectly.
 I will need to validate once I know how the `LAB_DATA.AMB` files work.